include/deal.II/distributed/cell_data_transfer.h

```diff
@@ -0,0 +1,346 @@
+     * Vector<double> data_to_transfer (triangulation.n_active_cells());
+     * //[fill data_to_transfer with cell by cell values...]
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * cell_data_trans.prepare_for_coarsening_and_refinement(data_to_transfer);
+     *
+     * // actually execute the refinement,
+     * triangulation.execute_coarsening_and_refinement();
+     *
+     * // unpack transferred data,
+     * Vector<double> transferred_data (triangulation.n_active_cells());
+     * cell_data_trans.unpack(transferred_data);
+     *
+     * @endcode
+     *
+     *
+     * <h3>Use for serialization</h3>
+     *
+     * This class can be used to serialize and later deserialize a distributed
+     * mesh with attached data to separate files. If you use more than one
+     * parallel::distributed::CellDataTransfer object, they need to be
+     * serialized and deserialized in the same order.
+     *
+     * For serialization, the following code snippet saves not only the
+     * triangulation itself, but also the cell by cell data attached:
+     * @code
+     * Vector<double> data_to_transfer (triangulation.n_active_cells());
+     * //[fill data_to_transfer with cell by cell values...]
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * cell_data_trans.prepare_serialization(data_to_transfer);
+     *
+     * triangulation.save(filename);
+     * @endcode
+     *
+     * Later, during deserialization, both triangulation and data can be
+     * restored as follows:
+     * @code
+     * //[create coarse mesh...]
+     * triangulation.load(filename);
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * Vector<double> transferred_data (triangulation.n_active_cells());
+     * cell_data_trans.deserialize(transferred_data);
+     * @endcode
+     *
+     * @note If you use more than one object to attach data to a
+     * parallel::distributed::Triangulation at the same time (e.g. a
+     * parallel::distributed::SolutionTransfer object) with the aim of
+     * serialization, the calls to
+     * parallel::distributed::CellDataTransfer::prepare_serialization()
+     * and parallel::distributed::SolutionTransfer::prepare_serialization(),
+     * as well as parallel::distributed::CellDataTransfer::deserialize() and
+     * parallel::distributed::SolutionTransfer::deserialize(),
+     * need to be in the same order, respectively.
+     *
+     * @note See documentation of parallel::distributed::SolutionTransfer for
+     * matching code snippets for both transfer and serialization.
+     *
+     * @ingroup distributed
+     * @author Marc Fehling, 2018
+     */
+    template <int dim, int spacedim = dim, typename VectorType = Vector<double>>
```

Comment:
You use a vector `typename VectorType` here. Can you comment in the class description how you assume this vector to be ordered (and what the underlying cell order is)? (For example, it would be equally possible to store cell-related data in a `std::map` with an active cell iterator as key.)

Reply:
The order, as in many other places, is the order in which active cells are traversed.

Reply:
I know, but it would be good to mention this here again.